### ISSUE TITLE
feat: add deep inheritance support

### ISSUE THREAD

**Issue by jpsav** [GillianPerard/typescript-json-serializer#64]
Thanks for creating typescript-json-serializer I have found it very useful ! 

I have a use case where I have multi level inheritance e.g. class B extends A and class C extends B where all the classes declare some properties that are annotated with  ` @JsonProperty` when calling `serialize` on an instance of class C the resulting JSON data only contains the properties declared on class C and B but not the properties declared on class A.  I would expect the resulting JSON data to contain the fields marked as Json Properties on all parent classes.

Example: 

```Typescript
@Serializable()
abstract class A {
    @JsonProperty()
    a: string;
}

@Serializable("A")
abstract class B extends A {
    @JsonProperty()
    b: string;
}

@Serializable("B")
class C extends B {
    @JsonProperty()
    c: string
}

const instance: C = new C();
instance.a = "aaa";
instance.b = "bbb";
instance.c = "ccc";

const data: any = serialize(instance);
/**
 * data = {
 *     b: "bbb",
 *     c: "ccc"
 * }
 * 
 * I expect data to be
 * {
 *     a: "aaa",
 *     b: "bbb",
 *     c: "ccc"
 * }
 */
```

Is it possible to update typescript-json-serializer to handle multiple level inhertance and include annotated properties from all parent classes ? 

Thanks for your time.

JPSav.

**Comment by GillianPerard** [comment]
Hi jpsav!, Thanks a lot to use my lib.

Let me check what is the problem!

**Comment by GillianPerard** [comment]
You're right, deep inheritance and multiple inheritance are missing from my lib, I add it as quickly I can!

**Comment by GillianPerard** [comment]
I've just published the new release 2.0.0, hope it fixes your issue!

**Comment by jpsav** [comment]
Thanks @GillianPerard !
